tests/lib/rules/no-unsafe.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @fileoverview Prevent usage of UNSAFE_ methods
+ * @author Sergei Startsev
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-unsafe');
+const RuleTester = require('eslint').RuleTester;
+
+const parserOptions = {
+  ecmaVersion: 2018,
+  sourceType: 'module',
+  ecmaFeatures: {
+    jsx: true
+  }
+};
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({parserOptions});
+ruleTester.run('no-unsafe', rule, {
+  valid: [
+    {
+      code: `
+        class Foo extends React.Component {
+          componentDidUpdate() {}
+          render() {}
+        }
+      `
+    },
+    {
+      code: `
+        const Foo = createReactClass({
+          componentDidUpdate: function() {},
+          render: function() {}
+        });
+      `
+    },
+    {
+      code: `
+        class Foo extends Bar {
+          UNSAFE_componentWillMount() {}
+          UNSAFE_componentWillReceiveProps() {}
+          UNSAFE_componentWillUpdate() {}
+        }
+      `
+    },
+    {
+      code: `
+        const Foo = bar({
+          UNSAFE_componentWillMount: function() {},
+          UNSAFE_componentWillReceiveProps: function() {},
+          UNSAFE_componentWillUpdate: function() {},
+        });
+      `
+    }
+  ],
+
+  invalid: [
+    {
+      code: `
+      class Foo extends React.Component {
+        UNSAFE_componentWillMount() {}
+        UNSAFE_componentWillReceiveProps() {}
+        UNSAFE_componentWillUpdate() {}
+      }
+    `,
+      errors: [
+        {
+          message: 'Do not use UNSAFE_componentWillMount'
```

Comment:
these tests should include position information, so that we can test that part as well

Reply:
Added a few more tests and position details for invalid cases

Reply:
It seems that added position details causes issues in eslint 3, see `TEST=true ESLINT=3` failed travis builds.

Reply:
Adjusted tests to support ESLint 3